src/metpy/io/gempak.py

```diff
@@ -0,0 +1,2830 @@
+        return stations
+
+    def _unpack_standard(self, sfcno):
+        """Unpack a standard surface data file."""
+        stations = []
+        for irow, row_head in enumerate(self.row_headers):
+            for icol, col_head in enumerate(self.column_headers):
+                if (irow, icol) not in sfcno:
+                    continue
+                station = {'STID': col_head.STID,
+                           'STNM': col_head.STNM,
+                           'SLAT': col_head.SLAT,
+                           'SLON': col_head.SLON,
+                           'SELV': col_head.SELV,
+                           'STAT': col_head.STAT,
+                           'COUN': col_head.COUN,
+                           'STD2': col_head.STD2,
+                           'SPRI': col_head.SPRI,
+                           'DATE': row_head.DATE,
+                           'TIME': row_head.TIME,
+                           }
+                for iprt, part in enumerate(self.parts):
+                    pointer = (self.prod_desc.data_block_ptr
+                               + (irow * self.prod_desc.columns * self.prod_desc.parts)
+                               + (icol * self.prod_desc.parts + iprt))
+                    self._buffer.jump_to(self._start, _word_to_position(pointer))
+                    self.data_ptr = self._buffer.read_int(4, self.endian, False)
+                    if not self.data_ptr:
+                        continue
+                    self._buffer.jump_to(self._start, _word_to_position(self.data_ptr))
+                    self.data_header_length = self._buffer.read_int(4, self.endian, False)
+                    data_header = self._buffer.set_mark()
+                    self._buffer.jump_to(data_header,
+                                         _word_to_position(part.header_length + 1))
+                    lendat = self.data_header_length - part.header_length
+
+                    if part.data_type == DataTypes.real:
+                        packed_buffer_fmt = '{}{}f'.format(self.prefmt, lendat)
+                        packed_buffer = (
+                            self._buffer.read_struct(struct.Struct(packed_buffer_fmt))
+                        )
+                    elif part.data_type == DataTypes.realpack:
+                        packed_buffer_fmt = '{}{}i'.format(self.prefmt, lendat)
+                        packed_buffer = (
+                            self._buffer.read_struct(struct.Struct(packed_buffer_fmt))
+                        )
+                    elif part.data_type == DataTypes.character:
+                        packed_buffer_fmt = '{}s'.format(lendat)
+                        packed_buffer = (
+                            self._buffer.read_struct(struct.Struct(packed_buffer_fmt))
+                        )
+                    else:
+                        raise NotImplementedError('No methods for data type {}'
+                                                  .format(part.data_type))
```

Comment:
```suggestion
                    fmt_code = {DataTypes.real: 'f', DataTypes.realpack: 'i', DataTypes.character: 's'}.get(part.data_type)
                    if fmt_code is None:
                        raise NotImplementedError('No methods for data type {}'
                                                  .format(part.data_type))
                    packed_buffer = (
                        self._buffer.read_struct(struct.Struct(f'{self.prefmt}{lendat}{fmt_code}'))
                    )
```

Reply:
Just a suggestion that *looks* like it would simplify things.

Reply:
Yeah, looks cleaner.